nilearn/reporting/glm_reporter.py

```diff
@@ -190,234 +188,138 @@ def make_glm_report(
+        smoothing_fwhm = None
+
+    model_attributes = model_attributes_to_dataframe(
+        model, is_volume_glm=is_volume_glm
     )
     with pd.option_context("display.max_colwidth", 100):
-        model_attributes = model_attributes_to_dataframe(model)
         model_attributes_html = dataframe_to_html(
             model_attributes,
             precision=2,
-            header=False,
+            header=True,
             sparsify=False,
         )
-    statistical_maps = make_stat_maps(model, contrasts)
-    html_design_matrices = _dmtx_to_svg_url(design_matrices)
-
-    # Select mask_img to use for plotting
-    if isinstance(model.mask_img, NiftiMasker):
-        mask_img = model.masker_.mask_img_
-    else:
-        try:
-            # check that mask_img is a niiimg-like object
-            check_niimg(model.mask_img)
-            mask_img = model.mask_img
-        except Exception:
-            mask_img = model.masker_.mask_img_
-
-    mask_plot_html_code = _mask_to_svg(
-        mask_img=mask_img, bg_img=bg_img, cut_coords=cut_coords
-    )
-    all_components = _make_stat_maps_contrast_clusters(
-        stat_img=statistical_maps,
-        contrasts_plots=contrast_plots,
-        threshold=threshold,
-        alpha=alpha,
-        cluster_threshold=cluster_threshold,
-        height_control=height_control,
-        two_sided=two_sided,
-        min_distance=min_distance,
-        bg_img=bg_img,
-        cut_coords=cut_coords,
-        display_mode=display_mode,
-        plot_type=plot_type,
-    )
-    all_components_text = "\n".join(all_components)
-    report_values_head = {
-        "page_title": escape(page_title),
-    }
-    report_values_body = {
-        "page_heading_1": page_heading_1,
-        "page_heading_2": page_heading_2,
-        "model_attributes": model_attributes_html,
-        "all_contrasts_with_plots": "".join(contrast_plots.values()),
-        "design_matrices": html_design_matrices,
-        "mask_plot": mask_plot_html_code,
-        "component": all_components_text,
-    }
-    report_text_body = report_body_template.safe_substitute(
-        **report_values_body
-    )
-    report_text = HTMLReport(
-        body=report_text_body,
-        head_tpl=report_head_template,
-        head_values=report_values_head,
-    )
-    # setting report size for better visual experience in Jupyter Notebooks.
-    report_text.width, report_text.height = check_report_dims(report_dims)
-    return report_text
 
+    contrasts = coerce_to_dict(contrasts)
 
-def _plot_to_svg(plot):
-    """Create an SVG image as a data URL \
-    from a Matplotlib Axes or Figure object.
-
-    Parameters
-    ----------
-    plot : Matplotlib Axes or Figure object
-        Contains the plot information.
-
-    Returns
-    -------
-    url_plot_svg : String
-        SVG Image Data URL.
-
-    """
-    try:
-        return figure_to_svg_quoted(plot)
-    except AttributeError:
-        return figure_to_svg_quoted(plot.figure)
```

Comment:
dumped this into a lower level function and made it more explicit because the try / catch here was just to handle save figures when the plot is an Axes and not a Figure